junit5-jupiter-extensions/src/test/java/com/example/cartesian/CartesianProductTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package com.example.cartesian;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.TestInfo;
+
+class CartesianProductTests {
+
+	@CartesianProductTest({"0", "1"})
+	void fourBits(String a, String b, String c, String d) {
+		int value = Integer.parseUnsignedInt(a + b + c + d, 2);
+		assertTrue((0B0000 <= value) && (value <= 0B1111));
```

Comment:
I think `0b0000` with a lower case `b` is a bit more readable, `B` looks very similar to `8`.